match2/commons/match_group_display_bracket.lua

```diff
@@ -1,738 +1,745 @@
-    self.winner = tonumber(match.winner)
-    self.dateexact = utils.misc.readBool(match.dateexact)
-    --self.date = Helper:parseDateString( match.date, self.dateexact )
-
-    -- add participants data from games to opponents
-    -- TODO: solve this for more than 2 opponents
-    local opponentParticipants = {{}, {}}
-    for index, game in ipairs(self.games) do
-        local participants =
-            type(game.participants) == 'table' and game.participants or Json.parse(game.participants or '{}')
-        local gameparticipants = {{}, {}}
-        for key, val in pairs(participants) do
-            local opPl = utils.string.split(key, '_')
-            if not gameparticipants[opPl[1]] then gameparticipants[opPl[1]] = {} end
-            gameparticipants[opPl[1]]['p' .. opPl[2]] = val
-        end
-        local gameName = 'g' .. (index + 1)
-        opponentParticipants[1][gameName] = gameparticipants[1]
-        opponentParticipants[2][gameName] = gameparticipants[2]
-    end
 
-    self.opponent1Raw.participants = opponentParticipants[1]
-    self.opponent2Raw.participants = opponentParticipants[2]
+        -- Distance between middle of match and top of round
+        local mid = headerFullHeight + matchMarginTop + matchHeight / 2
 
-    -- parse bracketdata
-    self.referencedIds = {}
-    self.type = bracketdata.type or 'error'
-    self.header = utils.misc.emptyOr(bracketdata.header or nil)
-    self.skipround = bracketdata.skipround or 0
-    if (self.skipround == 'true') then
-        self.skipround = 1
-    else
-        self.skipround = tonumber(self.skipround) or 0
-    end
-    self.qualskip = bracketdata.qualskip or 0
-    if (self.qualskip == 'true') then
-        self.qualskip = 1
-    else
-        self.qualskip = tonumber(self.qualskip) or 0
-    end
-    self.qualwin = utils.misc.readBool(bracketdata.qualwin or 0)
-    self.quallose = utils.misc.readBool(bracketdata.quallose or 0)
-    self.upper = utils.misc.emptyOr(bracketdata.toupper or nil)
-    self.lower = utils.misc.emptyOr(bracketdata.tolower or nil)
-    self.qualwinLiteral = utils.misc.emptyOr(bracketdata.qualwinLiteral or '')
-    self.qualloseLiteral = utils.misc.emptyOr(bracketdata.qualloseLiteral or '')
-    self.bracketreset = utils.misc.emptyOr(bracketdata.bracketreset or nil)
-    self.thirdplace = utils.misc.emptyOr(bracketdata.thirdplace or nil)
-    self.opponent1Raw.displaytype = 'bracket'
-    self.opponent2Raw.displaytype = 'bracket'
-
-    -- set bracket specific visual parameters
-    self.minDisplayHeight = 0
-    if (self.thirdplace ~= nil) and has3rd then
-        self.minDisplayHeight = 3 * self.matchHeight + 3 * self.padding + self.headerHeight + self.headerHeight2
-    elseif (self.quallose) then
-        self.minDisplayHeight = 3 * self.matchHeight + self.headerHeight
+        -- Height of this round, including the header but excluding the 3rd place match and qualifier rounds.
+        local height = headerFullHeight 
```

Comment:
And here

Reply:
> If you need to explain your variable with a comment you could probably give it a better name! 😉

While fully describing variables in their names is good practice in a lot of cases, I don't think it's a good aim in critical code paths, where variables are used to represent concepts that won't fit in a variable name. I think it's better to have variable names accurately represent its contents, but not a full representation. The variable name, type annotation, usage of the variable, scope, context, and comments all help describe the variable, so there's no reason to have the variable name do all the work. 

Here, all the layout fields are relative to a containing block, and different parts of the computation use different containing blocks. I opted to specify the containing block in the comments instead of the variable name. People working on this section of code will need to mentally track different containing blocks regardless of whether of it's in the variable name. So might as well keep the variable name short.

In this case the `height` field is only used to compute the layout of the higher round match it advances into. So for matches with 3rd place tiebreaks and qualifier spots, the `height` field is never used.